.github/workflows/test.yml

```diff
@@ -1,14 +1,21 @@
 name: CI
 
-on: [push]
+on:
+  pull_request:
+    branches:
+      - main
```

Comment:
This will automatically run ci for pr's merged into main.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpull_requestpull_request_targetbranchesbranches-ignore

Reply:
I would rather keep tests running for all branches; this is sometimes useful for more complex stacked PRs and Github pays for it, not us :)